src/DataSet/ObjectDataSet.php

```diff
@@ -61,41 +67,70 @@ public function getAttributeValue(string $attribute): mixed
             return $this->object->getAttributeValue($attribute);
         }
 
-        return isset($this->reflectionProperties[$attribute])
-            ? $this->reflectionProperties[$attribute]->getValue($this->object)
-            : null;
+        return $this->getData()[$attribute] ?? null;
     }
 
     public function getData(): array
     {
-        if ($this->dataSetProvided) {
-            return $this->object->getData();
+        $cacheItem = $this->getCacheItem();
+        if (isset($cacheItem['data'])) {
+            return $cacheItem['data'];
         }
 
-        $data = [];
-        foreach ($this->reflectionProperties as $name => $property) {
-            $data[$name] = $property->getValue($this->object);
+        if ($this->dataSetProvided) {
+            $data = $this->object->getData();
+        } else {
+            $data = [];
+            foreach ($this->reflectionProperties as $name => $property) {
+                $data[$name] = $property->getValue($this->object);
+            }
         }
+
+        $this->updateCacheItem('data', $data);
```

Comment:
Seems here may be collision when use several instances of `ObjectDataSet` for same object class. Maybe save `$data` to non-static private variable?

Reply:
What about using `spl_object_id` for the cache key.